gvm/utils.py

```diff
@@ -16,8 +16,69 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+# pylint: disable=arguments-differ, redefined-builtin
+
+import base64
 import warnings
+import logging
+
+from typing import Any, List
+from lxml import etree
+
+from gvm.xml import create_parser
+
+logger = logging.getLogger(__name__)
 
 
 def deprecation(message: str):
     warnings.warn(message, DeprecationWarning, stacklevel=2)
+
+
+def _check_command_status(xml: str) -> bool:
+    """Check gmp response
+
+    Look into the gmp response and check for the status in the root element
+
+    Arguments:
+        xml: XML-Source
+
+    Returns:
+        True if valid, otherwise False
+    """
+
+    if xml == 0 or xml is None:
+        logger.error("XML Command is empty")
+        return False
+
+    try:
+        root = etree.XML(xml, parser=create_parser())
+        status = root.attrib["status"]
+        return status is not None and status[0] == "2"
+
+    except etree.Error as e:
+        logger.error("etree.XML(xml): %s", e)
+        return False
+
+
+def _to_bool(value: bool) -> str:
+    return "1" if value else "0"
+
+
+def _to_base64(value: str) -> bytes:
+    return base64.b64encode(value.encode("utf-8"))
+
+
+def _to_comma_list(value: List) -> str:
+    return ",".join(value)
+
+
+def _add_filter(cmd, filter, filter_id):
+    if filter:
+        cmd.set_attribute("filter", filter)
+
+    if filter_id:
+        cmd.set_attribute("filt_id", filter_id)
+
+
+def _is_list_like(value: Any) -> bool:
+    return isinstance(value, (list, tuple))
```

Comment:
I am not sure about this function anymore. Maybe we should replace its usage with

```python
from collections.abc import Iterable

isinstance(value, Iterable)
```

or changing this function to `is_iterable`.

Reply:
Guess we found out, that its not 'that easy' :/